cmd/jaeger/internal/extension/remotesampling/extension.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package remotesampling
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	"github.com/jaegertracing/jaeger/pkg/clientcfg/clientcfghttp"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"github.com/jaegertracing/jaeger/plugin/sampling/leaderelection"
+	"github.com/jaegertracing/jaeger/plugin/sampling/strategystore/adaptive"
+	"github.com/jaegertracing/jaeger/plugin/sampling/strategystore/static"
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	"github.com/jaegertracing/jaeger/storage"
+	"github.com/jaegertracing/jaeger/storage/samplingstore"
+)
+
+var _ extension.Extension = (*rsExtension)(nil)
+
+const defaultResourceName = "sampling_store_leader"
+
+type rsExtension struct {
+	cfg              *Config
+	telemetry        component.TelemetrySettings
+	httpServer       *http.Server
+	grpcServer       *grpc.Server
+	strategyProvider strategystore.StrategyStore // TODO we should rename this to Provider, not "store"
+	adaptiveStore    samplingstore.Store
+	distLock         *leaderelection.DistributedElectionParticipant
+	shutdownWG       sync.WaitGroup
+}
+
+func newExtension(cfg *Config, telemetry component.TelemetrySettings) *rsExtension {
+	return &rsExtension{
+		cfg:       cfg,
+		telemetry: telemetry,
+	}
+}
+
+// GetAdaptiveSamplingStore locates the `remotesampling` extension in Host
+// and returns the sampling store and a loader/follower implementation, provided
+// that the extension is configured with adaptive sampling (vs. file-based config).
+func GetAdaptiveSamplingStore(
```

Comment:
@Pushkarm029 so I tried to experiment with cleaner separation of configs in #5635 and #5639, and came to the conclusion that it's not worth it, because there are at least 3 shared settings between aggregator and provider (provider introduced in #5634). It doesn't make sense to replicated those settings between extension and processor. It also doesn't make a lot of sense to separate adaptive sampling configuration between extension and processor, because it just increases mental overhead for the user. What I suggest we do instead is change this function to return this struct
```
package adaptive

type Components struct {
  Config *Config
  SamplingStore samplingstore.Store
  LeaderFollower *leaderelection.DistributedElectionParticipant 
}
```
This would be sufficient to instantiate aggregator (and post-processor)

I can take a stub at this change once we merge #5634